apps/speed.c

```diff
@@ -1778,7 +1778,12 @@ int speed_main(int argc, char **argv)
         buflen = lengths[size_num - 1];
         if (buflen < 36)    /* size of random vector in RSA benchmark */
             buflen = 36;
-        buflen += MAX_MISALIGNMENT + 1;
+        if (0x7fffffff - (MAX_MISALIGNMENT + 1) < buflen)
+        {
+            BIO_printf(bio_err, "Warning: ignoring -misalign option.\n");
+        } else {
```

Comment:
Because each branch of the _if_ is but a single statement, our coding style would suggest omitting the {} on both.

I think better might be changing the warning to an error and failing.  In which case, the {} should be retained.

Reply:
@paulidale Made it an error, also updated help with the maximum possible buffer size for clarity.

Reply:
Even better.